metals/src/main/scala/scala/meta/internal/metals/SemanticTokensProvider.scala

```diff
@@ -309,45 +309,48 @@ object SemanticTokensProvider {
       case _ => false
     }
 
-  def makeScalaCliTokens(
+  private def makeScalaCliTokens(
       comm: Token.Comment,
       initialDelta: Line,
   ): (List[Integer], Line) = {
-    val cliTokens = getTokens(false, comm.value)
+
+    val directive = comm.toString()
     val buffer = ListBuffer.empty[Integer]
+    val parts = directive.split("[,\\s]+").toList.zipWithIndex
     var delta = initialDelta
-    cliTokens.foreach { tk =>
-      val (toAdd, delta0) = tk match {
-        case start: Token.Ident if start.value == ">" =>
-          convertTokensToIntList(
-            "//>",
-            delta,
-            getTypeId(SemanticTokenTypes.Comment),
-          )
-        case using: Token.Ident if using.value == "using" =>
-          convertTokensToIntList(
-            using.value,
-            delta,
-            getTypeId(SemanticTokenTypes.Keyword),
-          )
-        case tk =>
-          val (tpe, mod) = typeModOfNonIdentToken(tk, false)
-          convertTokensToIntList(
-            tk.text,
-            delta,
-            tpe,
-            mod,
-          )
-      }
-      buffer.addAll(
-        toAdd
+    var text = directive
```

Comment:
So it's not really important but I feels a bit hard to follow how and why delta is changed in this manner.

For me it could be a bit more readable to use @tailrec function instead of foreach, so it's easier to see which args change and how they change. Also instead of `val idx = text.indexOf(part)` before `convertTokensToIntList` you could count the number of ` [,\\s]` after `convertTokensToIntList` to adjust the `text` and `delta` in one place.

I'm not sure it will make things much better so it's up to you to decide if you even want to do anything about it.

Reply:
I've  changed it to tailrec function, I agree its more readable , thanks for the suggestion!